### PR TITLE
Dynamo UI visual status of issues

### PR DESCRIPTION
**Pull request by SHKnudsen**
### Purpose

This pr adds a persistent notification when issues are present in the graph. The notification is always visible when a Graph Type is selected that has unresolved issues, even if the Graph Status extension is closed. If Graph Status extension is closed pressing the new notification will activate the Graph Status extension.

VisualStatusOfIssues

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@saintentropy @nate-peters @BogdanZavu 

### FYIs

@mjkkirschner @QilongTang

### CODE REVIEW

**Review by reddyashish — APPROVED** [review]
Looks good to me.

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -150,6 +150,15 @@ public WorkspaceModel CurrentSpace
             get { return model.CurrentWorkspace; }
         }
 
+        /// <summary>
+        /// Count of unresolved issues on the linter manager.
+        /// This is used for binding in the NotificationsControl
+        /// </summary>
+        public int LinterIssuesCount
+        {
+            get => Model.LinterManager.RuleEvaluationResults.Count;
```
Comment (saintentropy) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:159]: Null check?  Zero if null

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -661,6 +670,7 @@ protected DynamoViewModel(StartConfiguration startConfiguration)
            
             model.WorkspaceAdded += WorkspaceAdded;
             model.WorkspaceRemoved += WorkspaceRemoved;
+            model.LinterManager.RuleEvaluationResults.CollectionChanged += OnRuleEvaluationResultsCollectionChanged;
```
Comment (saintentropy) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:673]: Null check

Comment (saintentropy) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:673]: Also need to unsubscribe from the event

File: src/DynamoCoreWpf/Controls/Converters/LinterIssueCountToVisibilityConverter.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+using System.Windows;
+using System.Windows.Data;
+
+namespace Dynamo.UI.Controls.Converters
+{
+    /// <summary>
+    /// Converts an integer (linter issues count) to a visibility state
+    /// </summary>
+    [ValueConversion(typeof(int), typeof(Visibility))]
+    internal class LinterIssueCountToVisibilityConverter : IValueConverter
```
Comment (saintentropy) [inline src/DynamoCoreWpf/Controls/Converters/LinterIssueCountToVisibilityConverter.cs:16]: @SHKnudsen Can you move this to the DynamoCoreWpf\UI\Converters.cs class

Comment (saintentropy) [reply src/DynamoCoreWpf/Controls/Converters/LinterIssueCountToVisibilityConverter.cs:16]: I think that should also resolve the merge conflict

File: src/DynamoCoreWpf/Controls/NotificationsControl.xaml.cs [hunk 4]
```diff
@@ -44,5 +44,11 @@ void window_PreviewMouseDown(object sender, MouseButtonEventArgs e)
             // Commented this after MAGN - 8423
             // hsvm.ClearWarning();
         }
+
+        private void StackPanel_MouseDown(object sender, MouseButtonEventArgs e)
+        {
+            var viewModel = this.DataContext as DynamoViewModel;
+            viewModel.OnViewExtensionOpenRequest("Graph Status");
```
Comment (BogdanZavu) [inline src/DynamoCoreWpf/Controls/NotificationsControl.xaml.cs:51]: check for null viewModel ?

**Comment by BogdanZavu** [comment]
can we add some test as part DynamoCoreWpfTests ? Or if not part of this PR let's touch on this tomorrow

**Review by BogdanZavu — APPROVED** [review]

**Pull request merged**